packages/runtime/src/widgets/DataGrid/DataGrid.tsx

```diff
@@ -277,28 +287,42 @@ export const DataGrid: React.FC<GridProps> = (props) => {
   const onPageChangeAction = useEnsembleAction(onPageChange);
   // page change action
   const onPageChangeActionCallback = useCallback(
-    (page: number) => {
+    (page: number, newPageSize: number) => {
       if (onPageChangeAction) {
-        onPageChangeAction.callback({ page, pageSize: values?.pageSize });
+        onPageChangeAction.callback({
+          page,
+          pageSize: newPageSize,
+          totalRows: values?.totalRows,
+        });
       }
     },
     [onPageChangeAction],
   );
 
+  const storage = useEnsembleStorage();
+
   // handle page change
-  const handlePageChange = (page: number): void => {
-    onPageChangeActionCallback(page);
+  const handlePageChange = (page: number, newPageSize: number): void => {
+    storage.set(resolvedWidgetId, { page, pageSize: newPageSize });
```

Comment:
Let's remove this for now. We will handle the current page indexing and loading state separately.